chap10/10-8.rb

```diff
@@ -0,0 +1,139 @@
+# 9-4の日付クラスVer.1を以下のように改良せよ。
+# ・引数を受け取らないコンストラクタによるインスタンスの生成時は、西暦1年1月1日で初期化するのではなく、プログラム実行時の日付で初期化する。
+# ・引数を受け取るコンストラクタに不正な値が指定された場合は、適当な値に調整する。（たとえば、13月が指定された場合は12月とする/9月31日と指定された場合は9月30日とする。）
+# さらに、以下に示すに示すメソッドを追加すること
+# ある年が閏年であるかどうかを判定するクラスメソッド、日付が属する年が閏年であるかどうかを判断するメソッド、年内での経過日数（その年の元旦から数えて何日めであるか）を求めるメソッド、
+# 年内の残りの日数を求めるメソッド、他の日付との前後関係（より前の日付か/同じ日付か/より後ろの日付か）を判定するインスタンスメソッド、二つの日付の前後関係を判定するクラスメソッド、
+# 日付を一つ後ろに進めるメソッド（日付が2012年12月31日であれば、2013年1月1日に更新する）、次の日の日付を返却するメソッド、日付を一つ前に戻すメソッド、前の日の日付を返却するメソッド、
+# 日付をn日後ろに進めるメソッド、n日後の日付を返却するメソッド、日付をn日前に戻すメソッド、n日前の日付を返却するメソッドetc…
+
+require 'date'
+# class Dayは年月日それぞれに1を初期値として設定
+class Day
+    attr_accessor :year, :month, :date
+    # 平年と閏年の各月の日数を格納した配列。
+    @@normal_year = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+    @@leap_year = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
```

Comment:
クラス変数で管理せずとも定数で管理したほうが適切かなと思います。
```suggestion
    NORMAL_YEAR = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31].freeze
    LEAP_YEAR = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31].freeze
```